collect_app/src/main/java/org/odk/collect/android/widgets/SelectMultiWidget.java

```diff
@@ -150,6 +174,28 @@ protected void createLayout() {
         checkboxInit = false;
     }
 
+    protected View createWarning(String valuesWithSpaces) {
+        TextView warning = new TextView(getContext());
+
+        warning.setText(getContext().getResources().getString((valuesWithSpaces.contains(",") ?
+                R.string.invalid_space_in_answer_plural : R.string.invalid_space_in_answer_singular), valuesWithSpaces));
+
+        warning.setPadding(10, 10, 10, 10);
+        return warning;
+    }
+
+    protected String getValuesWithSpaces() {
+        StringBuilder valuesWithSpaces = new StringBuilder();
+        for (SelectChoice selectChoice : items) {
+            String value = selectChoice.getValue();
+            if (value.contains(" ")) {
+                valuesWithSpaces.append(value);
+                valuesWithSpaces.append(",");
+            }
+        }
+        return valuesWithSpaces.length() > 0 ? valuesWithSpaces.substring(0, valuesWithSpaces.length() - 1) : null;
+    }
+
```

Comment:
You can simplify this using `Joiner`. Notice also that I added a space after the comma (as in English).
```
        final List<String> valuesWithSpaces = new ArrayList<>();
        for (SelectChoice selectChoice : items) {
            String value = selectChoice.getValue();
            if (value.contains(" ")) {
                valuesWithSpaces.add(value);
            }
        }
        return valuesWithSpaces.isEmpty() ? null : Joiner.on(", ").join(valuesWithSpaces);
```

Reply:
I considered the `Joiner` but it needed an `ArrayList` so I opted for the `StringBuilder` which is sufficient for String manipulation. That said this looks good too.

Reply:
Glad to know you considered it. Since performance isn’t an issue, I’d go for using a method in a library that we already include. But it’s up to you.